sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images/sycl_ext_oneapi_bindless_images.asciidoc

```diff
@@ -0,0 +1,1826 @@
+                           const sycl::queue &queue);
+void *pitched_alloc_device(size_t *ret_pitch, const image_descriptor &desc, 
+                           const sycl::device &dev, const sycl::context &ctxt);
+
+void *pitched_alloc_device(size_t *ret_pitch, size_t width_in_bytes, 
+                           size_t height, unsigned int element_size_bytes, 
+                           const sycl::queue &queue);
+void *pitched_alloc_device(size_t *ret_pitch, size_t width_in_bytes, 
+                           size_t height, unsigned int element_size_bytes, 
+                           const sycl::device &dev, const sycl::context &ctxt);
+
+}
+```
+
+This function will allocate a memory region aimed to be used for 
+two-dimensional images. It allocates memory that is guaranteed to adhere to the 
+device's alignment requirements for 2D USM images.
+
+If the user does not wish to use `pitched_alloc_device` to allocate USM 2D 
+images, but prefers to use another USM allocation function instead, then that 
+allocation must adhere to some alignment restrictions. These restrictions are 
+device specific, and queries for them can be found in the "Pitch alignment 
+restrictions and queries" section below.
+
+If the allocation of pitched memory fails, `pitched_alloc_device` will throw a 
+`sycl::exception` with error code `sycl::errc::memory_allocation`.
+
+=== Pitch alignment restrictions and queries
+
+For the purposes of this document, the pitch of an image memory allocation is 
+the distance in bytes between the first elements of adjacent rows of the image.
+Some devices may require two-dimensional images to be allocated with specific 
+alignments for their width and pitch. The `pitched_alloc_device` API intends to 
+make allocation of USM memory adhering to these restrictions easy, returning an 
+appropriate pitch value to the user. However, if a user wishes to use another 
+USM allocation function, they must be aware of these restrictions, and query 
+the device to ensure the allocations they wish to use adhere to those 
+restrictions.
+
+This proposal provides a number of additional device queries that enable the 
+user to allocate appropriate USM memory for two-dimensional images. 
+
+The device information descriptors for these queries are:
+
+[frame="none",options="header"]
+|======================
+|Device descriptor |Return type |Description
+|`info::device::ext_oneapi_texture_pitch_align` |`uint32_t` | 
+Returns the required alignment of the pitch between two rows of an image in 
+bytes.
+|`info::device::ext_oneapi_max_texture_linear_width` |`size_t` | 
+Returns the maximum linear width allowed for images allocated using USM.
+|`info::device::ext_oneapi_max_texture_linear_height` |`size_t` | 
+Returns the maximum linear height allowed for images allocated using USM.
+|`info::device::ext_oneapi_max_texture_linear_pitch` |`size_t` | 
+Returns the maximum linear pitch allowed for images allocated using USM.
+|======================
```

Comment:
Same comment here that I made earlier about the namespace of device info descriptors.